In [1]:
import numpy as np
import pandas as pd
import requests
import json

In [2]:
subreddit = pd.read_csv('./workfile.csv')

In [3]:
subreddit.sub_name.values[25]

'OrlandoMagic'

In [32]:
df_sub = pd.read_csv('./data/Final/Android_final.csv')#.format(subreddit.sub_name.values[25]))

In [33]:
df_sub.head()

Unnamed: 0            author  \
0           0     AutoModerator   
1           1        LexiiBelle   
2           2            suda50   
3           3     anshumanpati6   
4           4  MalevolentFerret   

                                 comments_author_lst comments_score_lst  \
0                                                 []                 []   
1             ['thunder2132', 'AllGodsAreFalseGods']        [1416, 367]   
2      ['Jamesified', 'codenamejack', 'CosmosTruth']    [996, 616, 542]   
3                                                 []                 []   
4  ['BigAudioJackDongle', 'anshumanpati6', 'Limbo...   [1499, 974, 674]   

    created_utc  guilded  num_comments  pinned    r_id  score subreddit  \
0  1.538997e+09        0            10   False  9mdz1a      2   Android   
1  1.538939e+09        0           456   False  9m7sd9   2121   Android   
2  1.538940e+09        0           536   False  9m7waf    825   Android   
3  1.538970e+09        0            10   False  9mbgtd    137   Android   
4  1.538914e+09        0           470   False  9m4kqt   1813   Android   

                                               title  
0  Moronic Monday (Oct 08 2018) - Your weekly que...  
1  Companies should include wireless headphones f...  
2  OnePlus 6T: Removing the headphone jack was a ...  
3  POCO F1 gets a modified vendor for Android Pie...  
4  OnePlus on Twitter: "Check back in 24 hours fo...

In [34]:
df_sub.shape

(11600, 12)

In [35]:
df_sub.drop_duplicates('author',inplace=True)

In [36]:
auth_lst=df_sub.author.values

In [37]:
len(auth_lst)

396

- AskOuija start 4800 end 1580
- Futurology start 4800 end 126
- GameLive_0n start 4800 end 1
- Jokes start 4800 end 875
- NBA2k start 4800 end 961
- NoStupidQuestions start 4800 end 1801
- PoliticalHumor start 4800 end 638
- Showerthoughts start 4800 end 1881
- TwoXChromosomes start 4800 end 534
- europe starst 4800 end 393
- explainlikeimfive starts 4800 end 522
- funny start 4800 end 1525
- hiphopheads start 4800 end 702
- india start 4800 end 444
- lakers start 4800 end 264
- news start 4800 end 226
- nfl start 4800 end 443
- nottheonion start 4800 end 225
- opieandanthony start 4800 end 383
- science start 4800 end 177
- soccer start 4800 end 341
- technology start 4800 end 251
- timberwolves start 4800 end 270
- todayilearned start 4800 end 536
- videos start 4800 end 946
- yankees start 4800 end 242

In [45]:
sub_lst = []
for i in auth_lst:
    json_link  = 'https://www.reddit.com/user/{}/.json'.format(i)
    res = requests.get(json_link, headers={'User-agent': 'Capstone Bot 0.1'})
    data = res.json()
    #print(data)
    try:
        df=pd.DataFrame(data)
        df.head()
        try:
            df=pd.DataFrame(df.T.children[1])
            #print(len(df.data[1]))
        except:
            continue
        for i in range(24):
            if len(df.data[i])>=92:
                sub_lst.append(df.data[i]['subreddit'])
        print(sub_lst[-1])
    except: 
        continue
        
    

AskUK
AndroidGaming
Android
oneplus
GalaxyNote8
Android
Android
Android
Android
Android
Bitcoin
classicwow
GalaxyNote9
Android
microsoft
TheGoodPlace
metacanada
GooglePixel
DunderMifflin
gadgets
Android
arielwinter
hardwareswap
Android
Android
comics
Android
Android
Android
MicrosoftNews
GalaxyNote9
GalaxyNote9
apple
apple
lgv30
oneplus
Nokia
Nokia
PUBGvideos
Android
steelseries
GalaxyNote9
GalaxyNote9
Huawei
Android
Android
depression
Android
GooglePixel
GooglePixel
GooglePixel
Eyebleach
Malware
The_Donald
Showerthoughts
tmobile
fossdroid
SoundbarCircleJerk


KeyboardInterrupt: 

In [11]:
df = pd.DataFrame({'subreddits':sub_lst})

In [12]:
df.to_csv('./data/{}_posters_lst.csv'.format(subreddit.sub_name.values[25]),index=False)

In [31]:
    json_link  = 'https://www.reddit.com/user/{}/.json'.format(i)
    res = requests.get(json_link, headers={'User-agent': 'Capstone Bot 0.1'})
    data = res.json()
    #print(data)

    df=pd.DataFrame(data)
    df.head()
    df=pd.DataFrame(df.T.children[1])
    len(df.data[3])

94

In [13]:
#json_link  = 'https://www.reddit.com/user/YoungSandwich18/.json'

In [14]:
#res = requests.get(json_link, headers={'User-agent': 'Capstone Bot 0.1'})

In [15]:
#data = res.json()

In [16]:
#df=pd.DataFrame(data)

In [17]:
#df.T.children[1]

In [18]:
#df = pd.DataFrame(df.T.children[1])

In [19]:
#df.data[0]['subreddit']

In [20]:
#len(df.data[0])

In [21]:
#len(df.data[25]) #max is 24